MonoGame.Framework/Graphics/Effect/EffectParameter.cs

```diff
@@ -274,20 +276,20 @@ public Single[] GetValueSingleArray ()
 				}
 				return ret;
 			}
-			
-			switch(ParameterClass) 
-            {
-			case EffectParameterClass.Scalar:
-				return new Single[] { GetValueSingle () };
-            case EffectParameterClass.Vector:
-			case EffectParameterClass.Matrix:
-                    if (Data is Matrix)
-                        return Matrix.ToFloatArray((Matrix)Data);
-                    else
-                        return (float[])Data;
-			default:
-				throw new NotImplementedException();
-			}
```

Comment:
You may wish to split up minor things like indentation changes into its own commit separate from the functional changes.